python/pyspark/context.py

```diff
@@ -406,6 +406,22 @@ def defaultMinPartitions(self):
         """
         return self._jsc.sc().defaultMinPartitions()
 
+    @property
+    def numCores(self):
+        """
+        Total number of CPU cores of all executors registered in the cluster at the moment.
+        The number reflects current status of the cluster and can change in the future.
+        """
+        return self._jsc.sc().numCores()
+
+    @property
+    def numExecutors(self):
+        """
+        Total number of executors registered in the cluster at the moment.
+        The number reflects current status of the cluster and can change in the future.
```

Comment:
Shell we just manually add `.. versionadded:: 2.4.0`?

Reply:
I added it